### ISSUE TITLE
Endnotes heading in PDF is always H3

### ISSUE THREAD

**Issue by MichSpieler** [pressbooks/pressbooks#4276]
When a user chooses to display footnotes as endnotes in their PDF export, the "Notes" heading will always be assigned an H3. This is probably always the wrong heading level based on a chapter's structure.

<img width="851" height="508" alt="Image" src="https://github.com/user-attachments/assets/fd01d363-46f5-4881-ab9b-b39f32198fe1" />

Some thoughts on what heading level it should be:
- If the chapter does not have any other headings, then the endnotes heading should probably be H2.
- If the chapter does have headings for sub-sections, the endnotes heading should have the same level as the highest level used to make sense structurally. 
   -  If the author uses H1 (i.e. to use the two-level ToC feature), then "Notes" should also be H1. This could then lead to the issue that the "Notes" heading would appear in the ToC, which probably is not desired. So this should be prevented.
  - If the author does not use H1 but does use H2s in the chapter, then "Notes" should also be H2.

**Comment by tw77** [comment]
Possible increment to consider: making it H2 by default would already be an improvement

(Considering context / page content would be technically feasible, though)

**Comment by SteelWagstaff** [comment]
I agree with @tw77 H2 would be a good sensible default for now.

**Comment by arzola** [comment]
https://github.com/pressbooks/pressbooks/pull/4384

**Comment by MichSpieler** [comment]
I agree with the suggested intermediate solution to make it an H2. This will fix it for part of our users. It remains structurally wrong for the books that use multiple H1 headings in chapters. But I think that can be changed when we address this larger issue of multiple H1 elements.

**Comment by fdalcin** [comment]
Works as expected on dev. They are now rendered as H2 elements.
